docs/collection.md

```diff
@@ -311,3 +329,308 @@ $unsorted = new ArrayList(['x', 'c', 'a', 't', 'm']);
+
+```php
+<?php declare(strict_types=1);
+use phootwork\collection\Set;
+
+$fruits = new Set(['apple', 'pear', 'peach', 'pear', 'banana', 'apple']);
+
+$fruits->toArray(); // ['apple', 'pear', 'peach', 'banana']
+```
+
+### Addition
+
+You can add one or more elements to the *Set* via the [add()](api/phootwork/collection/Set.html#method_add) method:
+
+```php
+<?php declare(strict_types=1);
+use phootwork\collection\Set;
+
+$fruits = new Set(['apple', 'pear', 'peach']);
+
+$fruits->add('banana');
+$fruits->add('pineapple', ['kiwi', 'lychee'], 'dorian');
+
+$fruits->toArray(); // ['apple', 'pear', 'peach', 'pineapple', ['kiwi', 'lychee'], 'dorian']
+
+// Since the Set elements are unique, the instruction below doesn't modify the collection
+$fruits->add('apple', 'peach');
+
+$fruits->toArray(); // ['apple', 'pear', 'peach', 'pineapple', ['kiwi', 'lychee'], 'dorian']
+```
+
+!!! note
+    Set doesn't have the [insert()](#addition) method.
+    
+### More Methods
+
+Many Set methods are shared with other collections: please see the [api documentation](api/phootwork/collection/Set.html)
+and the examples above, in [ArrayList](#arraylist) section of this document.
+
+## Map
+
+A [Map](api/phootwork/collection/Map.html) is a collection of key-value pairs, where the value can be any type you want, objects included. You can think about Map
+as an associative array with super-powers.
+
+As usual, you can instantiate a new *Map* object by passing an associative array or an
+[Iterator](https://www.php.net/manual/en/class.iterator.php) object to the constructor:
+
+```php
+<?php declare(strict_types=1);
+use phootwork\collection\Map;
+
+$vehicles = new Map(['1 wheel' => 'unicycle', '2 wheels' => 'scooter', '4 wheels' => 'car']);
+```
+
+### Additions
+
+You can add one element to a Map via the [set()](api/phootwork/collection/Map.html#method_set) method, which requires
+the key as first argument and the value to add as second argument: 
+
+```php
+<?php declare(strict_types=1);
+use phootwork\collection\Map;
+
+$vehicles = new Map(['1 wheel' => 'unicycle', '2 wheels' => 'scooter', '4 wheels' => 'car']);
+
+$vehicles->set('3 wheels', 'sidecar');
+```
+
+You can add more elements via the [setAll()](api/phootwork/collection/Map.html#method_setAll) method. It expects an associative
+array or an [Iterator](https://www.php.net/manual/en/class.iterator.php) of elements to add:
+
+```php
+<?php declare(strict_types=1);
+use phootwork\collection\Map;
+
+$vehicles = new Map();
+
+$coll = [
+    '1 wheel' => 'unicycle',
+    '2 wheels' => 'scooter',
+    '3 wheels' => 'sidecar',
+    '4 wheels' => 'car'
+];
+
+$vehicles->setAll($coll);
+```
+
+### Getting the Elements
+
+You can get an element of the collection via the [get()](api/phootwork/collection/Map.html#method_get) method. It expects
+the key of the element and you can also pass a default value, to return if the element is not found:
```

Comment:
I think we should remove the default value here now, since php7 has the nullish coalescing syntax for that and rather document that one, eg.:

```php
$vehicles->get('6 wheels') ?? 'not found';
```

Reply:
Fixed.